src/docker/dockerfileParser.ts

```diff
@@ -0,0 +1,73 @@
+import * as fs from 'fs';
+import { CommandEntry, parse as rawDockerfileParser} from 'docker-file-parser';
+
+import { IDockerParser } from "./parser";
+import { shell, ShellResult } from "../shell";
+
+class RawDockerfile {
+    private commandEntries: CommandEntry[];
+
+    constructor(private readonly dockerfilePath: string) {
+        const dockerData = fs.readFileSync(dockerfilePath, 'utf-8');
+        this.commandEntries = rawDockerfileParser(dockerData, { includeComments: false });
+    }
+
+    public getCommandsOfType(...commands: string[]): CommandEntry[] {
+        return this.commandEntries.filter((entry) => {
+            const cmdName = entry.name.toLowerCase();
+            return commands.find((key) => key === cmdName);
+        });
+    }
+
+    public mergeCommandArgsOfType(command: string): string[] {
+        let args = [];
+        this.commandEntries.forEach((entry) => {
+            if (entry.name.toLowerCase() === command) {
+                args = args.concat(entry.args);
+            }
+        });
+        return args;
+    }
+
+    public searchInArgs(regularExpression: RegExp, scope?: string[]): RegExpMatchArray {
+        const scopeEntries = (scope ? this.getCommandsOfType(...scope) : this.commandEntries);
+        for (const entry of scopeEntries) {
+            const args = Array.isArray(entry.args) ? entry.args : [ String(entry.args) ];

```

Comment:
The use of `concat` in `mergeCommandsOfType` made it look like `entry.args` was always an array, but here you test it first.  Should it also be tested in `mergeCommandsOfType`?

Unit tests for these methods would be very reassuring.  I'm not sure how you're testing them at the moment so I don't know if I'm worrying about nothing or if it's stuff you've checked and proved to work.

Reply:
No need to test it in `mergeCommandsOfType`, because `concat([1])` equals `concat(1)`.

I have test the logic multiple times by running the real application, it worked. For command `FROM`, the arg is just a string, for `EXPOSE/ENTRYPOINT etc.`, the args is array.  Will consider to add some unit tests later.